Don't panic if we can't find the test plans

We should construct the engine as-needed instead of constructing it on init then panicing when we can't find the source dir:

```
attempting to guess testground source directory; for better control set ${TESTGROUND_SRCDIR}
panic: unable to determine testground src dir

goroutine 1 [running]:
github.com/ipfs/testground/cmd.glob..func1(0xc0001f4340)
	/cmd/all.go:12 +0x5b
github.com/ipfs/testground/cmd.init()
	/cmd/all.go:15 +0x37
```

--- a/cmd/all.go
+++ b/cmd/all.go
@@ -1,18 +1,25 @@
 package cmd
 
 import (
+	"sync"
+
 	"github.com/ipfs/testground/pkg/engine"
 	"github.com/urfave/cli"
 )
 
 // _engine is the default engine shared by all commands.
-var _engine *engine.Engine = func() *engine.Engine {
-	e, err := engine.NewDefaultEngine()
-	if err != nil {
-		panic(err)
-	}
-	return e
-}()
+var (
+	_engine     *engine.Engine
+	_engineErr  error
+	_engineOnce sync.Once
+)
+
+func GetEngine() (*engine.Engine, error) {
+	_engineOnce.Do(func() {
+		_engine, _engineErr = engine.NewDefaultEngine()
+	})
+	return _engine, _engineErr
+}
 
 // Commands collects all subcommands of the testground CLI.
 var Commands = []cli.Command{

--- a/cmd/build.go
+++ b/cmd/build.go
@@ -4,6 +4,7 @@ import (
 	"errors"
 
 	"github.com/ipfs/testground/pkg/api"
+	"github.com/ipfs/testground/pkg/engine"
 	"github.com/ipfs/testground/pkg/util"
 
 	"github.com/davecgh/go-spew/spew"
@@ -11,14 +12,9 @@ import (
 )
 
 var builders = func() []string {
-	b := _engine.ListBuilders()
-	if len(b) == 0 {
-		panic("no builders loaded")
-	}
-
-	names := make([]string, 0, len(b))
-	for k := range b {
-		names = append(names, k)
+	names := make([]string, 0, len(engine.AllBuilders))
+	for _, b := range engine.AllBuilders {
+		names = append(names, b.ID())
 	}
 	return names
 }()
@@ -60,12 +56,17 @@ func buildCommand(c *cli.Context) error {
 		builder = c.Generic("builder").(*EnumValue).String()
 	)
 
+	engine, err := GetEngine()
+	if err != nil {
+		return err
+	}
+
 	in, err := parseBuildInput(c)
 	if err != nil {
 		return err
 	}
 
-	out, err := _engine.DoBuild(plan, builder, in)
+	out, err := engine.DoBuild(plan, builder, in)
 	if err != nil {
 		return err
 	}

--- a/cmd/describe.go
+++ b/cmd/describe.go
@@ -39,7 +39,11 @@ func describeCommand(c *cli.Context) error {
 		return errors.New("unrecognized format for term;" + termExplanation)
 	}
 
-	plan := _engine.TestCensus().PlanByName(pl)
+	engine, err := GetEngine()
+	if err != nil {
+		return err
+	}
+	plan := engine.TestCensus().PlanByName(pl)
 	if plan == nil {
 		return errors.New("plan not found")
 	}

--- a/cmd/list.go
+++ b/cmd/list.go
@@ -14,7 +14,11 @@ var ListCommand = cli.Command{
 }
 
 func listCommand(c *cli.Context) error {
-	plans := _engine.TestCensus().ListPlans()
+	engine, err := GetEngine()
+	if err != nil {
+		return err
+	}
+	plans := engine.TestCensus().ListPlans()
 	for _, tp := range plans {
 		for _, c := range tp.TestCases {
 			fmt.Println(tp.Name + "/" + c.Name)

--- a/cmd/run.go
+++ b/cmd/run.go
@@ -6,20 +6,16 @@ import (
 	"strings"
 
 	"github.com/ipfs/testground/pkg/api"
+	"github.com/ipfs/testground/pkg/engine"
 	"github.com/ipfs/testground/pkg/util"
 
 	"github.com/urfave/cli"
 )
 
 var runners = func() []string {
-	r := _engine.ListRunners()
-	if len(r) == 0 {
-		panic("no runners loaded")
-	}
-
-	names := make([]string, 0, len(r))
-	for k := range r {
-		names = append(names, k)
+	names := make([]string, 0, len(engine.AllRunners))
+	for _, r := range engine.AllRunners {
+		names = append(names, r.ID())
 	}
 	return names
 }()
@@ -64,6 +60,11 @@ func runCommand(c *cli.Context) error {
 		return errors.New("missing test name")
 	}
 
+	engine, err := GetEngine()
+	if err != nil {
+		return err
+	}
+
 	// Extract flags and arguments.
 	var (
 		testcase     = c.Args().First()
@@ -97,7 +98,7 @@ func runCommand(c *cli.Context) error {
 		}
 
 		// Trigger the build job.
-		out, err := _engine.DoBuild(comp[0], builderId, in)
+		out, err := engine.DoBuild(comp[0], builderId, in)
 		if err != nil {
 			return fmt.Errorf("error while building test plan: %w", err)
 		}
@@ -128,6 +129,6 @@ func runCommand(c *cli.Context) error {
 		Parameters:   parameters,
 	}
 
-	_, err = _engine.DoRun(comp[0], comp[1], runnerId, runIn)
+	_, err = engine.DoRun(comp[0], comp[1], runnerId, runIn)
 	return err
 }

